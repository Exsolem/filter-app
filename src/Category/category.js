import React from 'react';
import { NavLink, withRouter } from "react-router-dom";
import styled from 'styled-components';
import uuid from 'uuid'
import queryString from 'query-string'

const CategoryLi = styled.li`
  text-align: left;
  border: 1px solid black;
  width: 20vw;
  list-style-type: none;
  font-size: 1.75vw;
  padding-left: 1vmin;
  &:hover { 
    background-color: rgba(0, 107, 54, 0.6);
    text-decoration: none;
    color: #dfe3e6;
  };
`;
const Category = ({ category, getSearchValue, getSearchedProducts, getFilteredProducts, getCurrentCategory, location: { search } }) => {
  const getProducts = () => {
    getCurrentCategory(category);
    getFilteredProducts();
    getSearchedProducts();
  };
  if (search.length > 0) {
    let newSearch = queryString.parse(search);
    getSearchValue(newSearch.search);
  }
  const searchedProducts = () => {
    if (search.length > 0) {
      let newSearch = queryString.parse(search); 
      getSearchValue(newSearch.search);
    }
    getSearchedProducts();
  };

  return (
    <NavLink
      to={{
        pathname: `/${category}`,
        search: `${search}`,
      }}
      key={uuid()}
      activeStyle={{
        color: 'white',
        backgroundColor: 'rgba(0, 107, 54, 0.6)'
      }}
      style={{
        color: 'black',
        textDecoration: 'none',
        backgroundColor: 'rgba(86, 125, 130, 1)'
      }}
      onClick={getProducts}
      onLoad = { searchedProducts }
    >
      <CategoryLi>
        {category}
      </CategoryLi>
    </NavLink>)
};
export default withRouter(Category);