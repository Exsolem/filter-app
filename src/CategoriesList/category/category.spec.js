import React from 'react'
import {shallow} from 'enzyme'

import Category from './category'


describe('Category', () => {
    const mockGetCurrentCategory = jest.fn(),
        mockFilteredProducts = jest.fn(),
        mockGetSearchedProduct = jest.fn(),
        mockGetSearchValue = jest.fn();

    const props = {
        category: '',
        getCurrentCategory: mockGetCurrentCategory,
        getFilteredProducts: mockFilteredProducts,
        getSearchedProducts: mockGetSearchedProduct,
        getSearchValue: mockGetSearchValue,
        search: 'search'
    };

    describe('product initial', () => {
        const category = shallow(<Category {...props}/>);
        it('renders properly', () => {
            expect(category.debug()).toMatchSnapshot()
        });
    });

});