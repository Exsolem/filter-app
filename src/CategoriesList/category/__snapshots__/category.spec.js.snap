// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category product initial renders properly 1`] = `
"<NavLink to={{...}} activeStyle={{...}} style={{...}} onClick={[Function: getProducts]}>
  <CategoryLi />
</NavLink>"
`;
